test-haskell/cabal/cabal-3.2.1.0.ebuild

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CABAL_FEATURES="test-suite"
+CABAL_FEATURES+=" nocabaldep" # in case installed Cabal is broken
+inherit haskell-cabal
+
+MY_PN="Cabal"
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="A the test executable for dev-haskell/cabal"
+HOMEPAGE="https://www.haskell.org/cabal/"
+SRC_URI="https://hackage.haskell.org/package/${MY_P}/${MY_P}.tar.gz"
+
+PATCHES="${FILESDIR}/${PN}-3.2.1.0-disable-hackage-test.patch"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND=">=dev-haskell/fail-4.9:=[profile?] <dev-haskell/fail-4.10:=[profile?]
+	>=dev-haskell/mtl-2.1:=[profile?] <dev-haskell/mtl-2.3:=[profile?]
+	>=dev-haskell/parsec-3.1.13.0:=[profile?] <dev-haskell/parsec-3.2:=[profile?]
+	>=dev-haskell/semigroups-0.18.3:=[profile?] <dev-haskell/semigroups-0.20:=[profile?]
+	>=dev-haskell/text-1.2.3.0:=[profile?] <dev-haskell/text-1.3:=[profile?]
+	>=dev-lang/ghc-7.8.2:=
+"
+DEPEND="${RDEPEND}
+	=dev-haskell/cabal-3.2.1.0
+	test? ( >=dev-haskell/base-compat-0.11.0 <dev-haskell/base-compat-0.12
+		>=dev-haskell/base-orphans-0.6 <dev-haskell/base-orphans-0.9
+		>=dev-haskell/diff-0.4 <dev-haskell/diff-0.5
+		>=dev-haskell/integer-logarithms-1.0.2 <dev-haskell/integer-logarithms-1.1
+		>=dev-haskell/optparse-applicative-0.13.2.0 <dev-haskell/optparse-applicative-0.16
+		>=dev-haskell/quickcheck-2.13 <dev-haskell/quickcheck-2.14
+		>=dev-haskell/stm-2.4.5.0 <dev-haskell/stm-2.6
+		dev-haskell/tagged
+		>=dev-haskell/tar-0.5.0.3 <dev-haskell/tar-0.6
+		>=dev-haskell/tasty-1.2.3 <dev-haskell/tasty-1.4
+		>=dev-haskell/tasty-golden-2.3.1.1 <dev-haskell/tasty-golden-2.4
+		dev-haskell/tasty-hunit
+		dev-haskell/tasty-quickcheck
+		dev-haskell/temporary
+		>=dev-haskell/tree-diff-0.1 <dev-haskell/tree-diff-0.2 )
+"
+
+S="${WORKDIR}/${MY_P}"
+CABAL_CORE_LIB_GHC_PV="PM:8.10.1 PM:8.10.2"
+
+src_configure() {
+	# Since we're still on an older version of quickcheck, rev this down for the
+	# test suite. This doesn't affect the actual cabal library, as the
+	# dependency is strictly for the test-suite.
+	cabal_chdeps 'QuickCheck >= 2.14 && < 2.15' 'QuickCheck >=2.13 && < 2.14'
```

Comment:
It should not remove `2.15` from the list if it is working fine. Same for DEPENDs.

Reply:
We don't currently have 2.15 in the overlay, and it doesn't look like we can add it until `ghc 8.10` drops. Probably this will go away before this PR is finalized.

I see your point though, I should have just lowered the lower-bound and left the upper-bound as-is.